silverback/subscriptions.py

```diff
@@ -0,0 +1,150 @@
+import asyncio
+import json
+from enum import Enum
+from typing import AsyncGenerator, Dict, List, Optional
+
+from ape.logging import logger
+from websockets import client as ws_client
+
+
+class SubscriptionType(Enum):
+    BLOCKS = "newHeads"
+    EVENTS = "logs"
+
+
+class Web3SubscriptionsManager:
+    websocket_reconnect_max_tries: int = 3
+    rpc_response_timeout_count: int = 10
+    subscription_polling_time: float = 0.1  # secs
+
+    def __init__(self, ws_provider_uri: str):
+        # TODO: Temporary until a more permanent solution is added to ProviderAPI
+        if "infura" in ws_provider_uri:
```

Comment:
```suggestion
        if "infura" in ws_provider_uri and "ws/v3" not in ws_provider_uri:
```

And then I would remove the TODO above it on line 21.

Here is my thinking: 

1. adhoc providers may not use the plugin and thus still have the wrong URI
2. This allows both the released and unreleased versions of ape-infura to work